doc/platform_api/cmis_management_state_machine.md

```diff
@@ -0,0 +1,138 @@
+# CMIS management state machine HLD
+
+# High Level Design Document
+## Rev 0.1
+
+# Table of Contents
+- [1. Revision](#1-revision)
+- [2. Scope](#2-scope)
+- [3. Motivation](#3-motivation)
+- [4. Abbreviations](#4-abbreviations)
+- [5. Introduction](#5-introduction)
+- [6. HLD design](#6-hld-design)
+  - [6.1 The Requirement](#61-the-requirement)
+  - [6.2 CMIS Management Platform Support](#62-cmis-management-platform-support)
+  - [6.3 Modules Management Thread](#63-modules-management-thread)
+  - [6.4 General Module Detection Flow](#64-general-module-detection-flow)
+  - [6.5 Module Detection Flow States And Description](#65-module-detection-flow-states-and-description)
+  - [6.6 Module Detection Flow States And Functions](#66-module-detection-flow-states-and-functions)
+
+# 1. Revision
+| Rev | Date | Author | Change description |
+|:----------:|:----------:|:--------------:|:----------------------:|
+| 0.1 | 23/10/2023 | Doron Barashi | Initial version|
+| | | | |
+
+# 2. Scope
+This document provides high level design for CMIS management state machine flow
+
+# 3. Motivation
+Customers want to have complete flexibility and independence between switch vendors and optical cables.

```

Comment:
this section is not related to the state machine if the scope is the state machine only.
what we should probably refer to the description of the cmis host management and why we need to extend the state machine. it is not clear to me why this feature is needed.

in any case the wording is not well described. if we need to keep it i would like to suggest the following

SONiC CMIS host management is defined to allow user to intergate and qualify different modules from by different vendors and change parameters as needed without the need of upgrading the entire sw and to have run time change of the module parameters.

Reply:
np. changed it in the md file I will share via email